worlds/witness/player_logic.py

```diff
@@ -794,23 +843,150 @@ def determine_unrequired_entities(self, world: "WitnessWorld") -> None:
             # Jungle Popup Wall Panel
         }
 
+        # In panel hunt, all panels are game, so all panels need to be reachable (unless disabled)
+        if goal == "panel_hunt":
+            for entity_hex in is_item_required_dict:
+                if static_witness_logic.ENTITIES_BY_HEX[entity_hex]["entityType"] == "Panel":
+                    is_item_required_dict[entity_hex] = True
+
         # Now, return the keys of the dict entries where the result is False to get unrequired major items
         self.ENTITIES_WITHOUT_ENSURED_SOLVABILITY |= {
             item_name for item_name, is_required in is_item_required_dict.items() if not is_required
         }
 
-    def make_event_item_pair(self, entity_hex: str) -> Tuple[str, str]:
-        """
-        Makes a pair of an event panel and its event item
-        """
-        action = " Opened" if self.REFERENCE_LOGIC.ENTITIES_BY_HEX[entity_hex]["entityType"] == "Door" else " Solved"
+    def pick_panel_hunt_panels(self, world: "WitnessWorld"):
```

Comment:
A lot of sections in this algorithm are explained with a comment. That's usually a sign that it could be split into more methods, with explicit names, and it would be more readable and more maintainable